File: 0121-BestTimeToBuyAndSellStock/BestTimeToBuyAndSellStock.java
class Solution {
    public int maxProfit(int[] prices) {
        int min = prices[0];
        int diff = 0;
        
        for (int i = 0; i < prices.length; i++){
            int temp = prices[i] - min;
            if (temp > diff){
                diff = temp;
            }
            
            if (prices[i] < min){
                min = prices[i];
            }
        }
        
        return diff;
    }
}

File: 0121-BestTimeToBuyAndSellStock/BestTimeToBuyAndSellStock.py
class Solution(object):
    def maxProfit(self, prices):
        """
        :type prices: List[int]
        :rtype: int
        """
        
        min = prices[0]
        diff = 0
        
        for i in range(len(prices)):
            temp = prices[i] - min
            if temp > diff:
                diff = temp
            if prices[i] < min:
                min = prices[i]
                       
        return diff
            
        
